### PR TITLE
Add support for rstudio.cloud (to be renamed to posit.cloud)

### PR DESCRIPTION
**Pull request by jgutman**
### Intent

Addresses https://github.com/rstudio/rstudio-pro/issues/3605

### Approach

There are several goals here we wish to accomplish. 

To test this requires 1) latest version on main of rsconnect 2) rsconnect#619 resolved and merged on main 3) rstudio.cloud account has been opted in to the feature flag for publishing content, managed by the Hosted folks

1. In Global Options > Publishing > Connect, there should be an option to connect a new posit.cloud account, with the correct icons, dialog title, dialog text, and hyperlinks.
2. In Global Options > Publishing, when a rstudio.cloud account is selected, the buttons to disconnect or reconnect should be grayed out.
3. The icons showing in Global Options > Publishing for an rstudio.cloud account should be the new icons (slightly different cloud from shinyapps.io accounts).
4. When attempting to publish static content, such as a standard (non-shiny Rmd) or a plot, only Rstudio Connect and RPubs should be visible in the publishing wizard, and only rstudio connect accounts should be displayed in the account list.
5. When attempting to publish Shiny content (either a Shiny web app or an Rmd with shiny runtime), all connected accounts should be displayed, including rstudio.cloud, shinyapps.io, and rstudio connect.
6. When attempting to publish Plumber APIs, all rstudio.cloud and rstudio connect accounts should be displayed in the account list, and shinyapps.io accounts should be hidden.
7. The icons showing in the account list for rstudio.cloud should be the new icons.

### CODE REVIEW

File: src/cpp/session/modules/SessionRSConnect.R [hunk 1]
```diff
@@ -477,7 +477,7 @@
    tryCatch({
       # check for any non-shinyapps.io accounts
```
Comment (jgutman) [inline src/cpp/session/modules/SessionRSConnect.R:478]:
```suggestion
      # check for any non-cloud accounts
```

**Comment by jgutman** [comment]
Screenshots:

image
image
image
image
image

**Comment by timtmok** [comment]
When I login to rstudio.cloud, there's no Tokens section in Accounts. Is that part of the feature flag?

The link to the guide also redirects to the main dashboard after when logging in to rstudio.cloud. Does this mean the link to the guide will work after some other changes are deployed?

**Comment by jgutman** [comment]
No they haven't actually added it to the website yet, you have to just use your shinyapps.io credentials. But that was the copy they requested and am assuming will be added shortly.

Also the links they gave me are for posit.cloud, so I don't think they're actually live yet, but we assume they will be by time of release.

**Comment by jmcphers** [comment]
We also need to update the Connect logo as the `(o)` logo is very outdated (Connect hasn't used it in years -- maybe they never did). New logo is in https://github.com/rstudio/rstudio-pro/issues/3605.

**Comment by jgutman** [comment]
Oh, got it, to the hub/network image you uploaded? No one ever answered me what that one was for, will add those image files shortly

**Comment by jgutman** [comment]
Seems like there is some question about the connect logo, so I'm going to do that in a separate PR

**Pull request merged**